language/move-prover/interpreter/src/concrete/player.rs

```diff
@@ -0,0 +1,1357 @@
+            Ok(vec![packed])
+        }
+        Operation::Unpack(module_id, struct_id, ty_args) => {
+            if cfg!(debug_assertions) {
+                assert_eq!(typed_args.len(), 1);
+            }
+            let unpacked = handle_unpack(ctxt, *module_id, *struct_id, ty_args, typed_args[0]);
+            Ok(unpacked)
+        }
+        Operation::GetField(module_id, struct_id, ty_args, field_num) => {
+            if cfg!(debug_assertions) {
+                assert_eq!(typed_args.len(), 1);
+            }
+            let field = handle_get_field(
+                ctxt,
+                *module_id,
+                *struct_id,
+                ty_args,
+                *field_num,
+                typed_args[0],
+            );
+            Ok(vec![field])
+        }
+        Operation::BorrowField(module_id, struct_id, ty_args, field_num) => {
+            if cfg!(debug_assertions) {
+                assert_eq!(typed_args.len(), 1);
+            }
+            let (is_mut, _) = local_state.get_type(dsts[0]).get_ref_type();
+            let field = handle_borrow_field(
+                ctxt,
+                *module_id,
+                *struct_id,
+                ty_args,
+                *field_num,
+                is_mut,
+                srcs[0],
+                typed_args[0],
+            );
+            Ok(vec![field])
+        }
+        Operation::MoveTo(module_id, struct_id, ty_args) => {
+            if cfg!(debug_assertions) {
+                assert_eq!(typed_args.len(), 2);
+            }
+            handle_move_to(
+                ctxt,
+                *module_id,
+                *struct_id,
+                ty_args,
+                typed_args[1],
+                typed_args[0],
+                global_state,
+            )
+            .map(|_| vec![])
+        }
+        Operation::MoveFrom(module_id, struct_id, ty_args) => {
+            if cfg!(debug_assertions) {
+                assert_eq!(typed_args.len(), 1);
+            }
+            handle_move_from(
+                ctxt,
+                *module_id,
+                *struct_id,
+                ty_args,
+                typed_args[0],
+                global_state,
+            )
+            .map(|object| vec![object])
+        }
+        Operation::GetGlobal(module_id, struct_id, ty_args) => {
+            if cfg!(debug_assertions) {
+                assert_eq!(typed_args.len(), 1);
+            }
+            handle_get_global(
+                ctxt,
+                *module_id,
+                *struct_id,
+                ty_args,
+                typed_args[0],
+                global_state,
+            )
+            .map(|object| vec![object])
+        }
+        Operation::BorrowGlobal(module_id, struct_id, ty_args) => {
+            if cfg!(debug_assertions) {
+                assert_eq!(typed_args.len(), 1);
+            }
+            let (is_mut, _) = local_state.get_type(dsts[0]).get_ref_type();
```

Comment:
is_mut should be always true.

Reply:
At the end of the transformation, yes, `is_mut` should always be true, but before the eliminate-imm-ref passs, I think either mut or imm refs can be created?

Reply:
Nvm, I found that it does not make sense to execute the stackless bytecode after arbitrary passes in the pipeline. The program might not be correct. For example:
- after `ReachDefAnalysis`, we may have two `$t2 = move($t1); $t3 = move($t1);`
- before write-backs are instrumented, the modified local root is never updated...

So I removed the code that was intended for per-pass processing and only execute at the end of the transformation pipeline.